docs/proposals/18-shoot-CA-rotation.md

```diff
@@ -0,0 +1,195 @@
+reviewers:
+- 
+---
+
+# Automated Shoot CA rotation
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+- [Open Questions](#open-questions)
+
+## Summary
+
+This proposal outlines an on-demand multi-step approach to rotate all certificate authorities (CA) used in a Shoot cluster. This process includes creating new CAs, invalidating the old ones and recreating all certs signed by the CAs. 
+
+We propose to bundle the rotation of *all* CAs in the Shoot together as one trigerrable action. This includes the recreation and invalidation of the following CAs and all certs signed by them:
+
+- shoot kube-api-server CA
+- kubelet CA
+- Etcd CA
+- Front proxy CA
+- Metrics server CA
+- VPN CA
+
+Naturally not all certificates communicating with the `kube-api-server` are under control of gardener. An example for a gardener controlled cert is the kubelet client certificate used to communicate with the api server. An example for credentials not controlled by gardener are kubeconfigs or client certs requested via `CertificateSigningRequest`s by the shoot owner.
+
+We propose to use a two step approach to rotate CAs. The start of each phase is triggered by the shoot owner. In summary the **first phase** is used to create new CAs (for example the new api server and client CA). Then we make sure that all server and clients trust *both* old and new CA. Next we renew all client certs and server certs so they are now signed by the new CAs. This includes a node rollout in order to propagate the certs to kubelets and restart all pods. Afterwards the user needs to change their client credentials to trust both old and new apiserver CA.
+In the **second phase** we basically remove all trust to the old CA for server and clients. This does not include a node rollout but all pods using SAs will continue to trust the old CA until they restart. Also the user needs to exchange all their client credentials again to no longer trust the old CA.
+
+A detailed overview of all steps required for each phase is given in the [proposal](#proposal) section of this GEP.
+
+*Introducing a new client CA*
+
+Currently client certificates and api server certificate are signed by the same CA. We propose to create a seperate client CA when triggering the rotation. The client CA is used to sign certs of clients talking to the API Server.
+
+## Motivation
+
+If we face the challenge to invalidate a CA be it due to company policy or due to a security incident we need to basically manually recreate and replace all CAs and certificates. The process of rotating by hand is cumbersome and could lead to errors due to the many steps needing to be performed in the right order. By automating the process we want to create a way to securely and easily rotate shoot CAs.
+
+### Goals
+- Offer a trigerable and safe solution to rotate all CAs in a shoot cluster.
```

Comment:
...and I guess also all other "credentials" that are otherwise "static", internal/hidden or external/exposed, e.g. the current kubeconfig / static token file https://github.com/gardener/gardener/blob/master/docs/usage/shoot_operations.md#rotate-kubeconfig-credentials ?